packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -451,10 +453,102 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
       );
     }
 
-    const passwordPolicy: PasswordPolicy = {
+    let validFlag = true;
+
+    // Check length options.
+    let minLengthFlag = false;
+    let maxLengthFlag = false;
+    if (policy.customStrengthOptions.minPasswordLength) {
+      if (password.length >= policy.customStrengthOptions.minPasswordLength) {
+        minLengthFlag = true;
+      } else {
+        validFlag = false;
+      }
+    }
+    if (policy.customStrengthOptions.maxPasswordLength) {
+      if (password.length <= policy.customStrengthOptions.maxPasswordLength) {
+        maxLengthFlag = true;
+      } else {
+        validFlag = false;
+      }
+    }
+
+    // Check character options.
+    let char;
+    let lowerCaseFlag = false;
+    let upperCaseFlag = false;
+    let numberFlag = false;
+    let nonAlphaNumericFlag = false;
+    for (let i = 0; i < password.length; i++) {
+      char = password.charAt(i);
+      if (/^\d$/.test(char)) {
+        numberFlag = true;
+      } else if (/^[A-Z]$/.test(char)) {
+        upperCaseFlag = true;
+      } else if (/^[a-z]$/.test(char)) {
+        lowerCaseFlag = true;
+      } else if (policy.allowedNonAlphanumericCharacters.includes(char)) {
+        nonAlphaNumericFlag = true;
+      }
+    }
+
+    // Combine status into single isValid flag.
+    if (policy.customStrengthOptions.containsLowercaseLetter) {
+      validFlag &&= lowerCaseFlag;
+    }
+    if (policy.customStrengthOptions.containsUppercaseLetter) {
+      validFlag &&= upperCaseFlag;
+    }
+    if (policy.customStrengthOptions.containsNumericCharacter) {
+      validFlag &&= numberFlag;
+    }
+    if (policy.customStrengthOptions.containsNonAlphanumericCharacter) {
+      validFlag &&= nonAlphaNumericFlag;
+    }
+
+    // Only include properties that are required by the password policy.
+    const status: PasswordValidationStatus = {
+      isValid: validFlag,
+      ...(policy.customStrengthOptions.minPasswordLength && {
+        meetsMinPasswordLength: minLengthFlag
+      }),
```

Comment:
This allows `meetsMinPasswordLength` to only be defined in the status if `minPasswordLength` is defined in the policy. Same with the other properties, since we only want them to be defined if they are required by the policy.